Lib/test/test_free_threading/test_heapq.py

```diff
@@ -0,0 +1,279 @@
+        ]
+
+        def heappushpop_func(heap: list[int], pushpop_items: list[int]):
+            for item in pushpop_items:
+                popped_item = heapq.heappushpop(heap, item)
+                self.assertTrue(popped_item <= item)
+
+        self.run_concurrently(
+            worker_func=heappushpop_func,
+            args=(heap, pushpop_items),
+            nthreads=NTHREADS,
+        )
+        self.assertEqual(OBJECT_COUNT, len(heap))
+        self.assertTrue(self.is_min_heap_property_satisfied(heap))
+
+    def test_racing_heapreplace(self):
+        heap = list(range(OBJECT_COUNT))
+        shuffle(heap)
+        heapq.heapify(heap)
+
+        replace_items = [
+            randint(-OBJECT_COUNT, OBJECT_COUNT) for _ in range(OBJECT_COUNT)
+        ]
+
+        def heapreplace_func(heap: list[int], replace_items: list[int]):
+            for item in replace_items:
+                popped_item = heapq.heapreplace(heap, item)
+
+        self.run_concurrently(
+            worker_func=heapreplace_func,
+            args=(heap, replace_items),
+            nthreads=NTHREADS,
+        )
+        self.assertEqual(OBJECT_COUNT, len(heap))
+        self.assertTrue(self.is_min_heap_property_satisfied(heap))
+
+    def test_racing_heapify_max(self):
+        max_heap = list(range(OBJECT_COUNT))
+        shuffle(max_heap)
+
+        def heapify_max_func(max_heap: list[int]):
+            heapq.heapify_max(max_heap)
+
+        self.run_concurrently(
+            worker_func=heapify_max_func, args=(max_heap,), nthreads=NTHREADS
+        )
+        self.assertTrue(self.is_max_heap_property_satisfied(max_heap))
+
+    def test_racing_heappush_max(self):
+        max_heap = []
+
+        def heappush_max_func(max_heap: list[int]):
+            for item in range(OBJECT_COUNT):
+                heapq.heappush_max(max_heap, item)
+
+        self.run_concurrently(
+            worker_func=heappush_max_func, args=(max_heap,), nthreads=NTHREADS
+        )
+        self.assertTrue(self.is_max_heap_property_satisfied(max_heap))
+
+    def test_racing_heappop_max(self):
+        max_heap = list(range(OBJECT_COUNT))
+        shuffle(max_heap)
+        heapq.heapify_max(max_heap)
+
+        # Each thread pops (OBJECT_COUNT / NTHREADS) items
+        self.assertEqual(0, OBJECT_COUNT % NTHREADS)
+        per_thread_pop_count = OBJECT_COUNT // NTHREADS
+
+        def heappop_max_func(max_heap: list[int], pop_count: int):
+            local_list = []
+            for _ in range(pop_count):
+                item = heapq.heappop_max(max_heap)
+                local_list.append(item)
+
+            # Each local list should be sorted
+            self.assertTrue(self.is_sorted_descending(local_list))
+
+        self.run_concurrently(
+            worker_func=heappop_max_func,
+            args=(max_heap, per_thread_pop_count),
+            nthreads=NTHREADS,
+        )
+        self.assertEqual(0, len(max_heap))
```

Comment:
This one is super nitpicky, but `0 == len(max_heap)` feels weird and foreign. 
```suggestion
        self.assertEqual(len(max_heap), 0)
```

Could you address this for all the following cases too?